tooling/minimize/src/main.rs

```diff
@@ -120,10 +129,147 @@ macro_rules! show_error {
+
+    let use_panic_abort = {
+        let mut yes = false;
+        let mut it = args.iter();
+        while let Some(a) = it.next() {
+            if let Some(s) = a.to_str() {
+                if s == "--crate-name" {
+                    if let Some(next) = it.next().and_then(|o| o.to_str()) {
+                        if next == "panic_abort" {
+                            yes = true;
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        yes
+    };
+
+    // Inject the Rust flags
+    for arg in DEFAULT_ARGS {
+        args.push(arg.into());
+    }
+
+    // If we are building dependencies, inject the sysroot flag
+    if std::env::var_os("MINIMIZE_BUILD_DEPS").is_some() {
+        let sysroot_dir = setup_sysroot();
+        args.push(format!("--sysroot={}", sysroot_dir.display()).into());
+    }
+
+    args.push("-C".into());
+
+    if use_panic_abort {
+        args.push("panic=abort".into());
+    } else {
+        args.push("panic=unwind".into());
+    }
+
+    // Invoke the rust compiler
+    let status = Command::new("rustc")
+        .args(args)
+        .env_remove("RUSTC")
+        .env_remove("MINIMIZE_BE_RUSTC")
+        .env_remove("MINIMIZE_BUILD_DEPS")
+        .status()
+        .expect("failed to invoke rustc in custom sysroot build");
+
+    std::process::exit(status.code().unwrap_or(1));
+}
+
+fn setup_sysroot() -> PathBuf {
+    // Determine where to put the sysroot.
+    let sysroot_dir = get_sysroot_dir();
+    let sysroot_dir = sysroot_dir.canonicalize().unwrap_or(sysroot_dir); // Absolute path 
+
+    if sysroot_dir.join("lib").exists() {
+        return sysroot_dir;
+    }
+
+    // Determine where the rust sources are located.
+    let rust_src = {
+        // Check for `rust-src` rustup component.
+        let rustup_src = rustc_build_sysroot::rustc_sysroot_src(Command::new("rustc"))
+            .expect("could not determine sysroot source directory");
+        if !rustup_src.exists() {
+            show_error!("`rust-src` not found");
+        } else {
+            rustup_src
+        }
+    };
+    if !rust_src.exists() {
+        show_error!("given Rust source directory `{}` does not exist.", rust_src.display());
+    }
+    if rust_src.file_name().and_then(OsStr::to_str) != Some("library") {
+        show_error!(
+            "given Rust source directory `{}` does not seem to be the `library` subdirectory of \
+             a Rust source checkout.",
+            rust_src.display()
+        );
+    }
+
+    // Final check to make sure the directory creation was actually a success
+    std::fs::create_dir_all(&sysroot_dir)
+        .unwrap_or_else(|e| show_error!("create sysroot dir: {e}"));
+
+    let mut it = std::env::args();
+    let target = it
+        .by_ref()
+        .position(|a| a == "--target")
+        .and_then(|_| it.next())
+        .or_else(|| std::env::var("CARGO_BUILD_TARGET").ok())
```

Comment:
Why are you checking `CARGO_BUILD_TARGET` here? That's an env var cargo sets for its build scripts. I don't see how it makes sense here.